CARDS-2172: PREMs Rehab survey filters

sdumitriu opened the pull request.

sdumitriu committed:
CARDS-2172: PREMs Rehab survey filters

sdumitriu reviewed (commented):
Should the "Inpatient" discharges from any of the excluded departments, e.g. BC-DENTISTRY, still be eligible for the Inpatient survey?

sdumitriu commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -317,6 +317,15 @@
+    },
+    "io.uhndata.cards.prems.internal.importer.ConfiguredDiscardFilter~Discard-Rehab-ComplexContinuingCare":{
+      "priority": 60,
+      "conditions": ["CLINIC = /Survey/ClinicMapping/78840662", "PATIENT_CLASS <> Inpatient Rehab"]
```
I think we could put this condition directly in the mapper, they have to be both from `Rehab*` and have class `Inpatient Rehab.`

sdumitriu replied:
Actually that would change what happens to patients that are discharge from Toronto Rehab but are not rehab inpatients. What should happen in these cases?

- patient was discharged from Toronto Rehab but is not an `Inpatient Rehab`
- patient was discharged from Toronto Rehab, but from one of the blacklisted departments, e.g. `Nursing Clinic` or `Dentistry`
    - and is either Inpatient Rehab or just Inpatient (both cases are possible for a Dentistry discharge, as far as I've seen in the database)
- patient was discharge from another hospital (TGH | TWH), but has class `Inpatient Rehab`

sdumitriu replied:
> Actually that would change what happens to patients that are discharge from Toronto Rehab but are not rehab inpatients. What should happen in these cases?
> 
>     * patient was discharged from Toronto Rehab but is not an `Inpatient Rehab`
>     * patient was discharged from Toronto Rehab, but from one of the blacklisted departments, e.g. `Nursing Clinic` or `Dentistry`
>       
>       * and is either Inpatient Rehab or just Inpatient (both cases are possible for a Dentistry discharge, as far as I've seen in the database)
> 
>     * patient was discharge from another hospital (TGH | TWH), but has class `Inpatient Rehab`

Excluding OP Discharge, there don't seem to be any patients in these cases.

sdumitriu commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -317,6 +317,15 @@
+      "priority": 60,
+      "conditions": ["CLINIC = /Survey/ClinicMapping/78840662", "DISCH_DEPT_NAME in BC-AAC CLINIC; BC-DENTISTRY; LC-ASSISTIVE TECHNOLOGY; LC-CHIROPODY CLINIC; LC-ITB CLINIC; LC-NEUROPHYSIOLOGY CLINIC; LC-NURSING CLINIC; LC-PHYSIATRY; LC-PSYCHIATRY CLINIC; LC-ROBSON CLINIC; LC-SEATING CLINIC; LC-SKIN AND WOUND; LC-SPINAL CORD REHAB-OP-ALLIED HEALTH; RN-CHRONIC PAIN LEAP SERVICE; UC-DENTISTRY; UC-E ...
+    },
+    "io.uhndata.cards.prems.internal.importer.ConfiguredDiscardFilter~Discard-Rehab-ComplexContinuingCare":{
```
Rename

sdumitriu commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -317,6 +317,15 @@
+    },
+    "io.uhndata.cards.prems.internal.importer.ConfiguredDiscardFilter~Discard-Rehab-ComplexContinuingCare":{
+      "priority": 60,
+      "conditions": ["CLINIC = /Survey/ClinicMapping/78840662", "PATIENT_CLASS <> Inpatient Rehab"]
```
Need to map PATIENT_CLASS

sdumitriu committed:
[misc] PREMS Clarity mapping: Move columns in the subject they actually refer to

sdumitriu committed:
CARDS-2172: PREMs Rehab survey filters
Add mapping for the PATIENT_CLASS column

sashaandjic reviewed (approved):
EDITED: not tested - wrong PR ( I thought it was 2171)

sdumitriu committed:
CARDS-2172: PREMs Rehab survey filters
Also discard patients discharged to Psych Rehab

sdumitriu committed:
CARDS-2172: PREMs Rehab survey filters
Discard all events that are not explicitly marked as Inpatient or Emergency
Discard OP Discharge events, which are usually transfers between departments

marta- reviewed (approved):
🤞
